packages/widget/src/web-component.tsx

```diff
@@ -60,14 +69,10 @@ initializeSkipWidget();
 
 export default WebComponent;
 
-type Stringify<T> = {
-  [K in keyof T]?: string;
-};
-
 declare global {
-  namespace JSX {
-    interface IntrinsicElements {
-      [WEB_COMPONENT_NAME]: Stringify<WebComponentProps>;
-    }
+  interface SkipWidgetElement extends HTMLElement, WidgetProps {}
+
+  interface HTMLElementTagNameMap {
+    [WEB_COMPONENT_NAME]: SkipWidgetElement;
```

Comment:
![image](https://github.com/user-attachments/assets/ad1836b2-a0c8-4832-bc89-9a13f3e0f3f0)

extending `WidgetProps` means that you get type data for the element!

Reply:
actually, I can remove extending by HTMLElement and only use `WidgetProps` directly